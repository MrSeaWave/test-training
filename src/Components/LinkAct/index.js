import React, { Component } from 'react';

class LinkAct extends Component {
  constructor(props) {
    super(props);

    this.state = {
      class: 'normal',
    };
  }

  UNSAFE_componentWillReceiveProps(nextProps, nextContext) {
    this.setState({ class: 'willRceiveProps' });
  }

  render() {
    const { page, children } = this.props;
    return (
      <a className={this.state.class} href={page || '#'}>
        {children}
      </a>
    );
  }
}

export default LinkAct;
