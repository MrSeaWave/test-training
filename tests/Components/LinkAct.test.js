import React from 'react';
import { create, act } from 'react-test-renderer';
import LinkAct from '../../src/Components/LinkAct';

it('Link Act 方法改变class数据', () => {
  let root;
  act(() => {
    root = create(<LinkAct page="http://www.facebook.com">----- Facebook ----- </LinkAct>);
  });
  expect(root.toJSON()).toMatchSnapshot();

  // 有个问题，暂时没办法运行出来
  act(() => {
    root = root.update(<LinkAct page="http://www.baidu.com">------ BaiDu ------</LinkAct>);
  });

  console.log('rooot', root);
  expect(root.toJSON()).toMatchSnapshot();
});
