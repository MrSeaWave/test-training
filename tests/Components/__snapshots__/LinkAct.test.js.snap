// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link Act 方法改变class数据 1`] = `
<a
  className="normal"
  href="http://www.facebook.com"
>
  ----- Facebook ----- 
</a>
`;

exports[`Link Act 方法改变class数据 2`] = `
<a
  className="normal"
  href="http://www.baidu.com"
>
  ------ BaiDu ------
</a>
`;
